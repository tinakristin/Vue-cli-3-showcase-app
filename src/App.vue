<template>
  <div id="app">
    <div id="nav">
      <router-link to="/">Home</router-link> |
      <router-link to="/about">About</router-link>
    </div>
    <router-view/>
    <hr/>
    <div class="row footer">
      <i class="date pull-left">
        <font-awesome-icon icon="smile" aria-hidden="true"/> 23. april 2019
      </i>
      <i>Vue-cli-3 test applikasjon laget av Tina Kristin Jensen</i>
      <img src="./assets/Felix_the_cat.png"
           alt="Felix the cat" class="felix pull-right"/>
    </div>
  </div>
</template>

<script>
import Vue from 'vue';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faSmile } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';

Vue.component('font-awesome-icon', FontAwesomeIcon);
library.add(faSmile);

export default {};
</script>

<style lang="scss">
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;

  .footer {
    width: 100%;
    margin: 20px 0;
    .felix {
      padding-left: 10%;
      text-align: right;
      width: 100px;
    }
    .date {
      padding-right: 5%;
      text-align: left;
      width: 100px;
    }
  }
}
#nav {
  padding: 30px;
  a {
    font-weight: bold;
    color: #2c3e50;
    &.router-link-exact-active {
      color: #42b983;
    }
  }
}
</style>
