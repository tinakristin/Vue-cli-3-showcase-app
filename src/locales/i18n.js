import Vue from 'vue';
import VueI18n from 'vue-i18n';
import translation from './nb-NO/translation.json';

Vue.use(VueI18n);

export default new VueI18n({
  locale: 'no', // set locale
  fallbackLocale: 'no',
  messages: {
    no: translation, // set locale messages
  },
});
