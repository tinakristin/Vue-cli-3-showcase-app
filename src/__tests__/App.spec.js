import Vuex from 'vuex';
import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import i18n from '../locales/i18n';
import App from '../App.vue';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);
localVue.component('font-awesome-icon', FontAwesomeIcon);

describe('App', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isAuthenticated: true,
        showGlobalMessage: true,
      },
      actions: {
        setKeycloakUpdateToken: () => {},
      },
    });
  });

  it('matches the snapshot', () => {
    const wrapper = shallowMount(App, { localVue, i18n, store });
    wrapper.setProps({ updateToken: () => {} });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
