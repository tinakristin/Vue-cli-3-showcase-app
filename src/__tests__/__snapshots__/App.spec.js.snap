// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div id="app">
  <div id="nav">
    <router-link-stub to="/" tag="a" event="click">Home</router-link-stub> |
    <router-link-stub to="/about" tag="a" event="click">About</router-link-stub>
  </div>
  <router-view-stub name="default"></router-view-stub>
  <hr>
  <div class="row footer"><i class="date pull-left">
      <font-awesome-icon-stub icon="smile" aria-hidden="true"></font-awesome-icon-stub> 23. april 2019
    </i> <i>Vue-cli-3 test applikasjon laget av Tina Kristin Jensen</i> <img src="./assets/Marcello_logo_small.png" alt="Marcello logo" class="marcello pull-right"></div>
</div>
`;
